app/Platform/Controllers/SystemController.php

```diff
@@ -76,16 +83,112 @@ public function destroy(System $system): void
+        $loggedInUser = request()->user();
+        $this->gameListService->withTicketCounts = (
+            $loggedInUser !== null
+            && $loggedInUser->getPermissionsAttribute() >= Permissions::Developer
+        );
 
-        return view($this->resourceName() . '.games')
-            ->with('system', $system)
-            ->with('resource', $this->resourceName());
+        $validatedData = $request->validate([
+            'sort' => 'sometimes|string|in:console,title,achievements,points,leaderboards,players,tickets,progress,retroratio,-title,-achievements,-points,-leaderboards,-players,-tickets,-progress,-retroratio',
+            'filter.populated' => 'sometimes|string|in:yes,no,all',
+            'filter.status' => 'sometimes|string|in:all,unstarted,lt-beaten-softcore,gte-beaten-softcore,gte-beaten-hardcore,gte-completed,eq-mastered,eq-beaten-softcore-or-beaten-hardcore,any-softcore,revised',
+        ]);
+        $sortOrder = $validatedData['sort'] ?? 'title';
+        $filterOptions = [
+            'populated' => $validatedData['filter']['populated'] ?? 'yes',
+        ];
+        if (isset($validatedData['filter']['status'])) {
+            $filterOptions['status'] = $validatedData['filter']['status'];
+        }
+
+        $gameIds = $system->games()->get(['ID'])->pluck('ID')->toArray();
+        $totalUnfilteredCount = count($gameIds);
+
+        $this->gameListService->initializeUserProgress($loggedInUser, $gameIds);
+        $this->gameListService->initializeGameList($gameIds);
+
+        if (isset($filterOptions['populated'])) {
+            $this->gameListService->filterGameList(function ($game) use ($filterOptions) {
+                return $this->usePopulatedFilter($game, $filterOptions['populated']);
+            });
+        }
+
+        if (isset($filterOptions['status'])) {
+            $this->gameListService->filterGameList(function ($game) use ($filterOptions) {
+                return $this->gameListService->useGameStatusFilter($game, $filterOptions['status']);
+            });
+        }
+
+        $this->gameListService->mergeWantToPlay($loggedInUser);
+        $this->gameListService->sortGameList($sortOrder);
+
+        $availableSorts = $this->gameListService->getAvailableSorts();
+        $availableCheckboxFilters = [];
+        $availableRadioFilters = [
+            [
+                'kind' => 'populated',
+                'label' => 'Has achievements',
+                'options' => [
+                    'yes' => 'Yes',
+                    'no' => 'No',
+                    'all' => 'Either',
+                ],
+            ],
+        ];
+
+        $availableSelectFilters = [];
+        if ($loggedInUser) {
+            $availableSelectFilters = [
+                [
+                    'kind' => 'status',
+                    'label' => 'Status',
+                    'options' => [
+                        'all' => 'All games',
+                        'unstarted' => 'No achievements earned',
```

Comment:
I don't think it's specifically related to this PR, but when I apply the "No achievements earned" filter to my ArduBoy list, Ardynia disappears.

All Games:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/ce46a1be-d08c-4a48-aaed-00122b8ee550)

No achievements earned:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/ecbad23e-d128-43a6-8afe-e7eb2fc05204)

Reply:
Good catch - in latest I've hunted down the bug that was causing this problem.